client/extensions/zoninator/state/locks/reducer.js

```diff
@@ -0,0 +1,67 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer, keyedReducer } from 'state/utils';
+import {
+	ZONINATOR_REQUEST_LOCK,
+	ZONINATOR_REQUEST_LOCK_ERROR,
+	ZONINATOR_UPDATE_LOCK,
+} from '../action-types';
+
+const isRequesting = createReducer(
+	{},
+	{
+		[ ZONINATOR_REQUEST_LOCK ]: () => true,
+		[ ZONINATOR_REQUEST_LOCK_ERROR ]: () => false,
+		[ ZONINATOR_UPDATE_LOCK ]: () => false,
+	}
+);
```

Comment:
Note that the data layer now provides `isRequesting` information out of the box -- see #20038 for an example 😄